src/store/GlobalState.tsx

```diff
@@ -0,0 +1,44 @@
+import { createContext, useEffect, useState } from 'react';
+import { collection, query, onSnapshot } from 'firebase/firestore';
+
+import db from '../firebase';
+
+import { Expense } from '../expense';
+
+interface GlobalState {
+  expenses: Expense[];
+}
+
+const initialState: GlobalState = { expenses: [] };
+
+export const GlobalContext = createContext(initialState);
+
+export const GlobalProvider = ({ children }) => {
```

Comment:
i like that the classes now much clearer responsibilities. e.g. GlobalProvider now is "only" responsible for keeping the expense list and an expense is responsible to manage a single one expense. love it!